app/Http/Controllers/V2/Dashboard/ViewTreeRestorationGoalController.php

```diff
@@ -0,0 +1,182 @@
+            ->whereNotNull('due_at')
+            ->whereIn('site_id', $siteIds)
+            ->groupBy('year', 'month')
+            ->get()
+            ->toArray();
+    }
+
+    private function filterProjectIdsByType($projectIds, $type)
+    {
+        return collect($projectIds)->filter(function ($row) use ($type) {
+            return $row->type === $type;
+        })->pluck('id')->toArray();
+    }
+
+    private function treeCountByDueDate(array $projectIds)
+    {
+        $projects = Project::whereIn('id', $projectIds)->get();
+
+        return $projects->sum(function ($project) {
+            return $project->trees_planted_count;
+        });
+    }
+
+    private function treeCountPerPeriod($siteIds, $distinctDates, $totalTreesGrownGoal)
+    {
+        $treesUnderRestorationActual = [];
+        $totalAmount = 0;
+
+        foreach ($distinctDates as $date) {
+            $year = $date['year'];
+            $month = $date['month'];
+            $treeSpeciesAmount = 0;
+
+            $reports = SiteReport::whereIn('site_id', $siteIds)
+                ->whereNotIn('v2_site_reports.status', SiteReport::UNSUBMITTED_STATUSES)
+                ->whereYear('v2_site_reports.due_at', $year)
+                ->whereMonth('v2_site_reports.due_at', $month)
+                ->get();
+
+            foreach ($reports as $report) {
+                $treeSpeciesAmount += $report->treeSpecies()->where('collection', TreeSpecies::COLLECTION_PLANTED)->sum('amount');
+            }
+
+            $totalAmount = $totalTreesGrownGoal;
+
+            $formattedDate = Carbon::create($year, $month, 1);
+
+            $treesUnderRestorationActual[] = [
+                'dueDate' => $formattedDate,
+                'treeSpeciesAmount' => (int) $treeSpeciesAmount,
+                'treeSpeciesPercentage' => 0,
+            ];
+        }
+
+        foreach ($treesUnderRestorationActual as &$treeData) {
+            $percentage = ($totalAmount != 0) ? ($treeData['treeSpeciesAmount'] / $totalAmount) * 100 : 0;
+            $treeData['treeSpeciesPercentage'] = floatval(number_format($percentage, 3));
+        }
+
+        return $treesUnderRestorationActual;
+    }
+
+    private function getLatestDueDate($distinctDates)
+    {
+        $latestYear = 0;
+        $latestMonth = 0;
+
+        foreach ($distinctDates as $entry) {
+            $year = $entry['year'];
+            $month = $entry['month'];
+
+            if ($year > $latestYear || ($year == $latestYear && $month > $latestMonth)) {
+                $latestYear = $year;
+                $latestMonth = $month;
+            }
+        }
+
+        $latestDate = [
+            'year' => $latestYear,
+            'month' => $latestMonth,
+        ];
+
+        return $latestDate;
+    }
+
+    private function getAverageSurvival(array $projectIds)
+    {
+        $averageSurvivalRate = ProjectReport::whereIn('project_id', $projectIds)->avg('pct_survival_to_date');
```

Comment:
Please check with product: typically when we aggregate data from reports, we only want the ones that have been either approved, or at least submitted. So this would be something more like
```
ProjectReport::hasBeenSubmitted()->whereIn('project_id', $projectIds)->avg('pct_survival_to_date')
```
or maybe
```
ProjectReport::isStatus(ReportStatusStateMachine::APPROVED)-> ...
```

This will avoid averaging in values from incomplete or unapproved reports.

Reply:
ok, the filter of approved and submitted has been added

Reply:
added isApproved filter to projectReport